docs(author-social): add README for social links blocks (#4552)
Automattic/newspack-plugin: src/blocks/author-profile-social/README.md

# Author Social Links Block

> A Gutenberg block that displays the author's social media links and contact information as a row of icon links.

Reviewer: ```suggestion
A Gutenberg block that displays the author's social media links and contact information as a row of linked icons.
```
What do you think about tweaking this a bit?

Author: Done: [88b3ee6](https://github.com/Automattic/newspack-plugin/pull/4552/commits/88b3ee6bb86b066634807295b935ea1251d9aff0).

## Overview

> The Author Social Links block (`newspack/author-profile-social`) renders social media icons for the current author. It is designed exclusively as an inner block of the [Author Profile block](../../../../repos/newspack-blocks/src/blocks/author-profile/README.md) and appears only in block themes that support nested layout mode.

Author: Addressed in [6e190b2](https://github.com/Automattic/newspack-plugin/pull/4552/commits/6e190b2ba87930eff80bb9208ed74873cf755a9b).

> The Author Social Links block (`newspack/author-profile-social`) renders social media icons for the current author. It is designed exclusively as an inner block of the [Author Profile block](https://github.com/Automattic/newspack-blocks/tree/trunk/src/blocks/author-profile) and appears only in block themes that support nested layout mode.

Reviewer: ```suggestion
The Author Social Links block (`newspack/author-profile-social`) renders social media icons for the current author. It is designed exclusively as an inner block of the [Author Profile block](https://github.com/Automattic/newspack-blocks/tree/trunk/src/blocks/author-profile) and appears only in the nested layout mode of the block, available when using a block theme.
```

Author: Done: [88b3ee6](https://github.com/Automattic/newspack-plugin/pull/4552/commits/88b3ee6bb86b066634807295b935ea1251d9aff0).
